tests/phpunit/tests/test-translate-page-for-posts.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+class Translate_Page_For_Posts_Test extends PLL_UnitTestCase {
+	static $posts_en, $posts_fr;
+	public $structure = '/%postname%/';
+
+
+	static function wpSetUpBeforeClass() {
+		parent::wpSetUpBeforeClass();
+
+		self::create_language( 'en_US' );
+		self::create_language( 'fr_FR' );
+
+		require_once POLYLANG_DIR . '/include/api.php';
+		$GLOBALS['polylang'] = &self::$polylang;
+
+		// page for posts
+		self::$posts_en = $en = self::factory()->post->create( array( 'post_title' => 'posts', 'post_type' => 'page' ) );
+		self::$polylang->model->post->set_language( $en, 'en' );
+
+		self::$posts_fr = $fr = self::factory()->post->create( array( 'post_title' => 'articles', 'post_type' => 'page' ) );
+		self::$polylang->model->post->set_language( $fr, 'fr' );
+
+		self::$polylang->model->post->save_translations( $en, compact( 'en', 'fr' ) );
+	}
+
+	function setUp() {
+		parent::setUp();
+
+		global $wp_rewrite;
+
+		// switch to pretty permalinks
+		$wp_rewrite->init();
+		$wp_rewrite->extra_rules_top = array(); // brute force since WP does not do it :(
+		$wp_rewrite->set_permalink_structure( $this->structure );
+
+		self::$polylang->model->post->register_taxonomy(); // needs this for 'lang' query var
+
+		update_option( 'show_on_front', 'page' );
+		update_option( 'page_for_posts', self::$posts_en );
+
+		// go to frontend
+		self::$polylang = new PLL_Frontend( self::$polylang->links_model );
+		self::$polylang->init();
+	}
+
+	public function test_translate_page_for_posts_on_default_language() {
+		self::$polylang->curlang = self::$polylang->model->get_language( 'en' );
+
+		$return = self::$polylang->static_pages->translate_page_for_posts( get_option( 'page_for_posts' ) );
+
+		$this->assertEquals(self::$posts_en, $return );
+	}
+
+	public function test_translate_page_for_posts_on_secondary_language() {
+		self::$polylang->curlang = self::$polylang->model->get_language( 'fr' );
```

Comment:
I have this test failing in my environment.

In fact, all of your tests use the languages values that are stored in the cache during the [update_option( 'page_for_posts' )](https://github.com/polylang/polylang/blob/eb22bfdc585b9ee680e610b0715c12da77b3da8b/tests/phpunit/tests/test-translate-page-for-posts.php#L40) call, which happens before the [`PLL_Frontend_Static_Pages` is instantiated](https://github.com/polylang/polylang/blob/0c9fd403caa735875c2a5dad077948f8a03f1103/frontend/frontend.php#L68), so the `PLL_Model::get_languages_list()` call is not filtered by [PLL_Static_Pages::pll_languages_list()](https://github.com/polylang/polylang/blob/0c9fd403caa735875c2a5dad077948f8a03f1103/include/static-pages.php#L85-L94).